sympy/simplify/tests/test_trigsimp.py

```diff
@@ -410,6 +410,12 @@ def valid(a, b):
         assert s == exptrigsimp(e)
         assert valid(s, 2*sinh(a))
 
+def test_exptrigsimp_noncommutative():
+    a,b = symbols('a b', commutative=False)
+    x = Symbol('x', commutative=True)
+    assert exp(a + x) == exptrigsimp(exp(a)*exp(x))
+    p = exp(a)*exp(b) - exp(b)*exp(a)
+    assert p == exptrigsimp(p) != 0
```

Comment:
It seems  `assert p == exptrigsimp(p)` would serve the same purpose with fewer characters.

Reply:
The extra != 0 is probably nearly extraneous in that it is unlikely to fail without something else also failing, but I hoped its presence would clarify the purpose of the test, should it ever prove relevant.